core/src/main/java/tech/pegasys/eth2signer/core/signing/ArtifactSignature.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.core.signing;
+
+public interface ArtifactSignature {
```

Comment:
A _little_ concerned about this interface...

It implies a signature can only be represented by a hex string, and doesn't have "data" of its own - its actually more of a formatter, than an interface...

I.e. on our API we must define that a signature will be produced as a hex string, which has a specific formatting.
However - that formatting is captured in this class - _not_ as part of the interface, which doesn't feel very MVC - feels like we embedded interface knowledge in our business logic.

Maybe Eth2Signer is small enough that this doesn't matter,  but it _feels_ potentially dangerous.

Reply:
Got rid of this. Now returning the native Signature for whatever parser returns. Later we can add something which format this in another PR.